### PR TITLE
INSTUI-3066 - TS typing `ui-overlays`

### PR DESCRIPTION
**Pull request by ke1k0**

### CODE REVIEW

File: packages/ui-overlays/src/Mask/index.tsx [hunk 1]
```diff
@@ -23,10 +23,11 @@
  */
 
 /** @jsx jsx */
-import { Component } from 'react'
+import React, { Component } from 'react'
 // @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module 'no-s... Remove this comment to see the full error message
```
Comment (matyasf) [inline packages/ui-overlays/src/Mask/index.tsx:27]: you can get rid of this error by importing `"@types/no-scroll": "^2"` as a DevDependency

File: packages/ui-overlays/src/Mask/props.ts [hunk 2]
```diff
@@ -21,7 +21,8 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
  * SOFTWARE.
  */
-import React from 'react'
+
+import React, { InputHTMLAttributes } from 'react'
```
Comment (matyasf) [inline packages/ui-overlays/src/Mask/props.ts:25]: I get an 'import unused' error here

File: packages/ui-overlays/src/Mask/props.ts [hunk 3]
```diff
@@ -32,12 +33,15 @@ import type {
 import type { WithStyleProps, ComponentStyle } from '@instructure/emotion'
 
 type MaskOwnProps = {
+  children?: React.ReactNode
   onDismiss?: (...args: any[]) => any
   placement?: 'top' | 'center' | 'bottom' | 'stretch'
   fullscreen?: boolean
   onClick?: (...args: any[]) => any
```
Comment (matyasf) [inline packages/ui-overlays/src/Mask/props.ts:40]: These event handlers can be typed better

File: packages/ui-overlays/src/Overlay/props.ts [hunk 4]
```diff
@@ -28,153 +28,215 @@ import PropTypes from 'prop-types'
...
 
-type OverlayProps = OverlayOwnProps & OtherHTMLAttributes<OverlayOwnProps>
+  /**
+   * The type of `<Transition />` to use for animating in/out
+   */
+  transition?: TransitionType // passed as to Transition as `type`
 
-const propTypes: PropValidators<PropKeys> = {
-  children: PropTypes.node,
+  // TODO: deprecate applicationElement, it was removed from Dialog in v6
```
Comment (matyasf) [inline packages/ui-overlays/src/Overlay/props.ts:53]: Can you please make a ticket of this one? Or just fix here, as I see setting this value does nothing

**Comment by matyasf** [comment]
You forgot to remove 'shouldPrintComment' from babel.config.js

File: packages/ui-overlays/src/Mask/props.ts (hunk 3 above)
Comment (ke1k0) [reply packages/ui-overlays/src/Mask/props.ts:40]: Yeah, I'm sorry, I haven't finished this ticket yet but accidentally made a PR for it already (reflex I guess). There are stuff I want to marked as deprecated and I wanted to wait for https://github.com/instructure/instructure-ui/pull/856 to be merged before I do it. Sorry about the confusion and thanks for the review anyway :)

File: packages/ui-overlays/src/Mask/index.tsx (hunk 1 above)
Comment (ke1k0) [reply packages/ui-overlays/src/Mask/index.tsx]: Oh indeed, thanks for the tip!

File: packages/ui-overlays/src/Mask/props.ts (hunk 2 above)
Comment (ke1k0) [reply packages/ui-overlays/src/Mask/props.ts]: removed, thanks

File: packages/ui-overlays/src/Mask/props.ts (hunk 3 above)
Comment (ke1k0) [reply packages/ui-overlays/src/Mask/props.ts]: Done! Had to deprecate `onDismiss`, it hasn't been used in the component for a long time.

File: packages/ui-overlays/src/Overlay/props.ts (hunk 4 above)
Comment (ke1k0) [reply packages/ui-overlays/src/Overlay/props.ts]: Deprecated it.

**Review by matyasf — APPROVED** [review]

**Review by Brailor — APPROVED** [review]

**Pull request merged**